src/tool/migration-cli/src/main/java/org/niis/xroad/migration/signer/KeyConfMigrator.java

```diff
@@ -0,0 +1,231 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package org.niis.xroad.migration.signer;
+
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBElement;
+import jakarta.xml.bind.JAXBException;
+import jakarta.xml.bind.Unmarshaller;
+import lombok.extern.slf4j.Slf4j;
+import org.bouncycastle.crypto.generators.Argon2BytesGenerator;
+import org.bouncycastle.crypto.params.Argon2Parameters;
+import org.niis.xroad.migration.utils.DbCredentials;
+import org.niis.xroad.migration.utils.DbPropertiesReader;
+import org.niis.xroad.signer.keyconf.CertRequestType;
+import org.niis.xroad.signer.keyconf.CertificateType;
+import org.niis.xroad.signer.keyconf.DeviceType;
+import org.niis.xroad.signer.keyconf.KeyConfType;
+import org.niis.xroad.signer.keyconf.KeyType;
+import org.niis.xroad.signer.keyconf.ObjectFactory;
+
+import java.io.Console;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.sql.SQLException;
+import java.util.Objects;
+import java.util.Optional;
+
+@Slf4j
+public class KeyConfMigrator {
+
+    private static final String SOFT_TOKEN = "softtoken";
+
+    public void migrate(String keyconfPath, String dbPropertiesPath) throws SQLException {
+        KeyConfType keyConf = parseKeyConf(Path.of(keyconfPath, "keyconf.xml"));
+
+        SignerRepository repo = getSignerRepository(dbPropertiesPath);
+
+        for (DeviceType deviceType : keyConf.getDevice()) {
+            log.info("Processing token {}", deviceType.getId());
+            boolean isSoftToken = SOFT_TOKEN.equalsIgnoreCase(deviceType.getDeviceType());
+            long tokenId = handleToken(deviceType, repo, isSoftToken, keyconfPath);
+
+            log.info("  Processing token keys({})", deviceType.getKey().size());
+            for (KeyType keyType : deviceType.getKey()) {
+                log.info("    Processing key {}", keyType.getKeyId());
```

Comment:
I assume padding is for visibility? perhaps add each padding level as a constant?